pilot/pkg/status/reporter.go

```diff
@@ -69,10 +69,9 @@ const labelKey = "internal.istio.io/distribution-report"
 const dataField = "distribution-report"
 
 // Starts the reporter, which watches dataplane ack's and resource changes so that it can update status leader
-// with distribution information
-func (r *Reporter) Start(clientSet kubernetes.Interface, namespace string, store model.ConfigStore, stop <-chan struct{}) {
+// with distribution information.  To run in read-only mode, (for supporting istioctl wait), set writeMode = false
```

Comment:
Can you explain why this is needed? Enabling this by default, even in "read-only" mode is a red flag for me. Didn't we have `istioctl wait` long before this whole status reporter?

Reply:
Yes, but istioctl wait did not account for proxies which are skipped for updates.  In Istio 1.4, every change was distributed to every proxy, so there was no need to track skipping.  It looks like sometime in 1.5 we began skipping proxies for efficiency, which broke 'wait'.  

All this read-only mode does is track the skipped and acked versions of each proxy in a non-blocking map (and reverse map), so I expect the impact to be minimal.  Also, the addition of the integ test will keep this from being invisibly broken again.